src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -5284,6 +5284,13 @@ private StatementAst FunctionDeclarationRule(Token functionToken)
             // A function name that matches a keyword isn't really a keyword, so don't color it that way
             functionNameToken.TokenFlags &= ~TokenFlags.Keyword;
 
+            // A function name is not an expandable token, so remove any nested tokens
+            var expandableToken = functionNameToken as StringExpandableToken;
```

Comment:
Pattern matching makes this a little easier to read:

```c#
if (functionNameToken is StringExpandableToken expandableToken)
{
    expandableToken.NestedTokens = null;
}
```

Reply:
ahhh… that's what that suggestion message means?  I had copied that from a similar part in `GetCommandArgument`.

Reply:
I'm not sure what suggestion you're referring to, but pattern matching was not available in C# when the code you copied was written.

Reply:
I've addressed this code in revised commit, but have a ways to go on tests.

Reply:
VS Code suggests all over 'Use Pattern Matching'.  The area of the code I copied was not one such area though, probably because its too complex for the pattern matching syntax.